### DISCUSSION TITLE
Error when building Android: ':app:minifyReleaseWithR8'

### DISCUSSION THREAD

**Discussion by burekas7** [codemagic-ci-cd/codemagic-docs#2742] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hi,

I recently updated the Flutter version to 3.24.2
And when I try to build a version but I get an android build failure:

```
ERROR: Missing classes detected while running R8. Please add the missing classes or apply additional keep rules that are generated in /Users/builder/clone/build/app/outputs/mapping/release/missing_rules.txt.
ERROR: R8: Missing class com.google.j2objc.annotations.RetainedWith (referenced from: com.google.common.collect.ImmutableSet com.google.common.collect.ImmutableMap.entrySet and 3 other contexts)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete
```

On debug, everything is fine and working.

I tried putting # android.enableR8=true in the comment
(as it appears in another application since it is being built properly, and it still didn't help)

What could be the reason?

Build ID:
66e83d141a04c8ef79f8e6d0

**Comment by burekas7** [comment]
Solved:
I added this file: android/app/proguard-rules.pro

And added the needed rules according to the missing_rules.txt file.

**Reply by p-Osteen** [reply]
Can you give your 'proguard-rules.pro' file? Still facing this issue.
